### DISCUSSION TITLE
Postgres - Point (Postgis)

### DISCUSSION THREAD

**Discussion by ainsleyclark** [payloadcms/payload#8644] [Feature Requests & Ideas]
Now that Drizzle supports [Postgis geometry point](https://orm.drizzle.team/docs/guides/postgis-geometry-point), it would be fantastic to see it supported in the Postgres adapter.

**Comment by ainsleyclark** [comment]
I was thinking of something similar too:

1. **Config Update**: Add `enablePostgis` in the Postgres config, running `CREATE EXTENSION postgis` if true.
2. **Schema Changes**: Define `geometry(Point, 4326)` for Point fields and add spatial index support.
3. **Query Builder**: Extend `sanitizeQueryValue` to support near and other spatial operators using PostGIS functions `ST_DWithin`.
4. **Error Handling**: If `enablePostgis` isn’t enabled and Point fields are used, throw an error in queries.

I'm not sure what else would be required?

**Comment by ainsleyclark** [comment]
Anyone dropping in here, from @r1tsuu on Discord:

> This should be very soon! I already implemented a point field for postgres on [my branch](https://github.com/payloadcms/payload/compare/beta...feat/point-field-postgres) actually, but we won't merge it until this drizzle PR https://github.com/drizzle-team/drizzle-orm/pull/3141

**Comment by JesperWe** [comment]
Point support for Postgres has landed in v3 beta.

**Comment by ainsleyclark** [comment]
Amazing news, thank you so much! @r1tsuu
